src/v/security/authorizer.h

```diff
@@ -131,6 +124,42 @@ class authorizer final {
     }
 
 private:
+    /*
+     * Compute whether the implied operations are allowed based on the specified
+     * operation.
+     */
+    bool acl_any_implied_ops_allowed(
+      const acl_matches& acls,
+      const acl_principal& principal,
+      const acl_host& host,
+      const acl_operation operation) const {
+        auto check_op = [&acls, &principal, &host](acl_operation operation) {
+            return acls.contains(
+              operation, principal, host, acl_permission::allow);
+        };
+
+        switch (operation) {
+        case acl_operation::describe: {
+            std::array ops = {
```

Comment:
why not mark static?

Reply:
i suppose it may get inline anyway?

Reply:
That's what I assumed yes. 

Codegen is actually quite interesting: https://gcc.godbolt.org/z/vfd6Mfasn

Depending on the numbers clang actually generates some bithacks to shortcut. Also codegen can be different dependent on the numbers between using static and not. However, looking at llvm-mca (you need to comment out the functions you are not interested in or copy to https://uica.uops.info/) it seems the non-static version generates more efficient code.

Reply:
A local `static constexpr` usually seems the best to me, if it's possible: keeps the table local to the scope it is used, doesn't pose any chance of stack allocations a non-static array might, or the "lazy init" that a non-constexpr static might cause ([like this](https://gcc.godbolt.org/z/r3j7csYPY) though there has to be something that doesn't let the compiler do static init in the .data or .rodata sections for that to happen).

H/e oddly here `static` non-constexpr has different codegen from the other options as Stephan points odd. Pretty odd an inexplicable (which is itself not really odd weird when talking about compiler opts).

Reply:
Switched to `static constexpr`